src/app/transcript/[...slug]/page.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { allTranscripts } from "contentlayer/generated";
+import { LanguageCodes } from "@/config";
+import { notFound } from "next/navigation";
+import IndividualTranscript from "@/components/individual-transcript/IndividualTranscript";
+
+// forces 404 for paths not generated from `generateStaticParams` function.
+export const dynamicParams = false;
+
+export function generateStaticParams() {
+  const allSingleTranscriptPaths = allTranscripts.map((transcript) => {
+    const slugForLanguage = transcript.languageURL.split("/").filter(path => Boolean(path.trim()));
+    return {
+      slug: slugForLanguage
+    }
+  });
+
+  return allSingleTranscriptPaths;
+}
+
+const Page = ({ params }: { params: { slug: string[] } }) => {
+  const slugArray = params.slug;
+  const isNonEnglishLanguage = LanguageCodes.includes(slugArray[0]);
+  let transcriptUrl = "";
+  if (isNonEnglishLanguage) {
+    const languageCode = slugArray.shift();
+    slugArray[slugArray.length - 1] = slugArray[slugArray.length - 1] + `.${languageCode}`;
+    transcriptUrl = `/${slugArray.join("/")}`;
+  } else {
+    transcriptUrl = `/${slugArray.join("/")}`;
+  }
```

Comment:
Can we have a comment explaining what this is about?

Reply:
Good catch, this was a previous implementation that's no longer needed with implementation of the languageURL field in Contentlayer config for transcripts.
Regardless what this was meant to do was create a url that is similar to what is coming from `transcript.url` field in content layer, and for transcripts of other languages it returns it with the slug.

e.g `/zh/advancing-bitcoin/2022/2022-03-03-sanket-kanjalkar-miniscript`